### DISCUSSION TITLE
Import  FUI JavaScript in webpack

### DISCUSSION THREAD

**Discussion by MacLake** [fomantic/Fomantic-UI#2670] [General] [answered]
There are instructions around how to include the less version of FUI in webpack, and it’s also possible to build the normal FUI and include the resulting CSS files in webpack.

But how can I import the FUI JS file `semantic.js`  in webpack, so that it is available in  scripts in HTML pages that include this webpack bundle, i. e. not only scripts within the bundle. E . g. for jQuery and htmx I can do it like so:

```JavaScript
import $ from 'jquery'
import 'htmx.org'
import 'fomantic-ui/dist/semantic.css'

window.jQuery = $
window.$ = $
window.htmx = require('htmx.org')
```

How can I do something similar like `window.$ = $` for FUI?

**Comment by lubber-de** [comment]
fomantic is a jquery plugin (or a collection of plugins) and no es6 module and thus cannot be imported the es6 way but needs to be included via script tags right after jquery
Something webpack isnt really prepared/designed for.

maybe the Provide-plugin helps?
https://webpack.js.org/plugins/provide-plugin/#usage-jquery

A simple boilerplate is here https://github.com/fomantic/Fomantic-UI/discussions/2692#discussioncomment-4869814

**Comment by lubber-de** [comment, marked as answer]
Seems to work within webpack when requiring from the src/definitions folder
See https://github.com/fomantic/Fomantic-UI/discussions/2692#discussioncomment-4873606
